gsa/src/web/pages/extras/trashcanpage.js

```diff
@@ -119,105 +115,85 @@ const separateByUsageType = inputList => {
   return {scan, compliance};
 };
 
-class Trashcan extends React.Component {
-  constructor(...args) {
-    super(...args);
-    this.state = {
-      trash: undefined,
-      loading: false,
-    };
-
-    this.createContentRow = this.createContentRow.bind(this);
-    this.createContentsTable = this.createContentsTable.bind(this);
-    this.getTrash = this.getTrash.bind(this);
-    this.handleEmpty = this.handleEmpty.bind(this);
-    this.handleDelete = this.handleDelete.bind(this);
-    this.handleRestore = this.handleRestore.bind(this);
-    this.handleErrorClose = this.handleErrorClose.bind(this);
-  }
+const Trashcan = () => {
+  const gmp = useGmp();
+  const [, renewSessionTimeout] = useUserSessionTimeout();
 
-  componentDidMount() {
-    this.getTrash();
-  }
+  const [trash, setTrash] = useState({});
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState();
 
-  getTrash() {
-    const {gmp} = this.props;
+  const getTrash = useCallback(() => {
     const data = gmp.trashcan.get().then(
       response => {
         const trash = response.data;
-        this.setState({trash});
+        setTrash(trash);
       },
       error => {
-        this.setState({error});
+        setError(error);
       },
     );
     return data;
-  }
+  }, [gmp.trashcan]);
 
-  handleInteraction() {
-    const {onInteraction} = this.props;
-    if (isDefined(onInteraction)) {
-      onInteraction();
-    }
-  }
+  useEffect(() => {
+    getTrash();
+  }, [getTrash]);
 
-  handleRestore(entity) {
-    const {gmp} = this.props;
+  const handleInteraction = useCallback(() => {
+    renewSessionTimeout();
+  }, [renewSessionTimeout]);
 
-    this.handleInteraction();
+  const handleRestore = useCallback(
+    entity => {
+      handleInteraction();
 
-    return gmp.trashcan
-      .restore(entity)
-      .then(this.getTrash)
-      .catch(error => {
-        this.setState({
-          error: error,
+      return gmp.trashcan
+        .restore(entity)
+        .then(getTrash)
+        .catch(error => {
+          setError(error);
         });
```

Comment:
Can be simplified if you want.
```suggestion
        .catch(setError);
```